### ISSUE TITLE
Enhancement: [prefer-optional-chain] Support simple ternary expressions

### ISSUE THREAD

**Issue by boris-petrov** [typescript-eslint/typescript-eslint#7859]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-optional-chain/

### Description

`prefer-optional-chain` could also support cases like the following:

```ts
return this.foo == null ? undefined : this.foo.bar;
```

I believe this is a (very) common pattern, i.e. any ternary check where the check is `== null` and the "then" clause is `undefined` and the "else" clause uses the same thing as in the check.

### Fail

```typescript
return this.foo == null ? undefined : this.foo.bar;
```

### Pass

```typescript
return this.foo?.bar;
```

**Comment by natanielmendes** [comment]
This issue has already been solved!

My testing project has the following versions in the `package.json`:
"@typescript-eslint/eslint-plugin": "^6.13.1",
 "@typescript-eslint/parser": "^6.13.1",
"eslint": "^8.54.0",

That's my `.eslintrc.json` which was used to test this linting validation:
```
{
  "root": true,
  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
  "parser": "@typescript-eslint/parser",
  "parserOptions": { "project": ["./tsconfig.json"] },
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": [
      2,
      {
        "allowString": false,
        "allowNumber": false
      }
    ],
    "@typescript-eslint/prefer-optional-chain": "error",
    "@typescript-eslint/no-unused-variable": "error"
  },
  "ignorePatterns": ["src/**/*.test.ts", "src/frontend/generated/*"]
}
```

Double-check whether this issue is still present even after upgrading your `typescript-eslint` to the latest version. 

Please close this Issue if the lint check is not failing anymore.

**Comment by natanielmendes** [comment]
@boris-petrov I have this implementation and I'm not getting any `error` or `warn`:
```
export class AppComponent implements OnInit {
  title = 'typescript-eslint-785';
  private readonly foo: { bar: number } | undefined = { bar: 10 };

  ngOnInit(): void {
    this.bar();
  }

  public bar(): number | undefined {
    return this.foo == null ? undefined : this.foo.bar;
  }
}
```

So that's why I think this issue should be closed, as it's not happening anymore in the latest version.
